plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/DependencyVersionsFetcher.Companion.kt

```diff
@@ -15,32 +15,31 @@ internal operator fun DependencyVersionsFetcher.Companion.invoke(
 ): DependencyVersionsFetcher? {
     val group = dependency.group ?: return null // TODO: Support NPM dependencies from Kotlin/JS
     val name = dependency.name
-    when (repository.url.scheme) {
-        "https", "file" -> Unit // Supported
-        "http" -> return null //TODO: Show non fatal error that http is not supported for security reasons.
-        else -> return null //TODO: Support more transport protocols. Here's what Gradle supports:
-        // https://docs.gradle.org/current/userguide/declaring_repositories.html#sec:supported_transport_protocols
-        // We should trigger a warning that it's not supported yet, with link to relevant issue,
-        // and we should report any transport protocol not known to be supported by Gradle (but not crash
-        // in case a future Gradle version adds more of them and we're not updated yet to support it).
-    }
-    return MavenDependencyVersionsFetcher(
-        httpClient = httpClient,
-        moduleId = ModuleId(group, name),
-        repoUrl = repository.url.toString().let { if (it.endsWith('/')) it else "$it/" },
-        repoAuthorization = when (repository.name) {
-            "MavenLocal" -> null // MavenLocal never needs authorization.
-            // Getting credentials on mavenLocal results in insanely
-            // hard to  diagnose error that happens later with an unrelated stacktrace.
-            // See this issue: https://github.com/jmfayard/refreshVersions/issues/222
-            // The error message would be the following:
-            // "Authentication scheme 'all'(Authentication) is not supported by protocol 'file'"
-            else -> repository.passwordCredentials?.let { credentials ->
+    return when (repository.url.scheme) {
+        "https" -> MavenDependencyVersionsFetcherHTTP(
+            httpClient = httpClient,
+            moduleId = ModuleId(group, name),
+            repoUrl = repository.url.toString().let { if (it.endsWith('/')) it else "$it/" },
+            repoAuthorization = repository.passwordCredentials?.let { credentials ->
                 Credentials.basic(
                     username = credentials.username ?: return@let null,
                     password = credentials.password ?: return@let null
                 )
             }
-        }
-    )
+        )
+        "file" -> MavenDependencyVersionsFetcherFILE(
+            moduleId = ModuleId(group, name),
+            repoUrl = repository.url.toString()
+        )
+        "gcs" -> MavenDependencyVersionsFetcherGCS(
```

Comment:
Doesn't follow Kotlin official naming conventions either.
We'll use `MavenDependencyVersionsFetcherGoogleCloudStorage` to avoid leaving domain specific acronyms in the codebase that might confuse future contributors for a bit.